pkgs/tools/graphics/mangohud/default.nix

```diff
@@ -194,16 +195,26 @@ stdenv.mkDerivation (finalAttrs: {
     ''}
   '';
 
-  postFixup = ''
+  postFixup = let
+    archMap = {
+      "x86_64-linux" = "x86_64";
+      "i686-linux" = "x86";
+    };
+    layerPlatform = archMap."${stdenv.hostPlatform.system}" or null;
+    # We need to give the different layers separate names or else the loader
+    # might try the 32-bit one first, fail and not attempt to load the 64-bit
+    # layer under the same name.
+  in lib.optionalString (layerPlatform != null) ''
+    substituteInPlace $out/share/vulkan/implicit_layer.d/MangoHud.${layerPlatform}.json \
+      --replace "VK_LAYER_MANGOHUD_overlay" "VK_LAYER_MANGOHUD_overlay_${toString stdenv.hostPlatform.parsed.cpu.bits}"
```

Comment:
I don't have the means to test but if MangoApp needs this too, the file should be modified here aswell when support is enabled.

cc @Scrumplex

Reply:
Yep this should be done for MangoApp too

Reply:
Oh wait sorry nvm!

Reply:
Right now our current build isn't even supporting mixed 32bit & 64bit mangoapp. Since it runs in a separate process, there's no need to have a 32bit build for overlaying 32bit games.